### PR TITLE
Replace Effect Updates

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR adds "replace first..." to the replace effect. I based this off of the current replace effect. Along with this, I added the ability to mark the replacements as case sensitive (so you aren't forced to specify case sensitivity in the config) as I thought it could come in use for some users.

I have tested the replace first effect and it worked fine. However, I have not yet tested the case sensitivity part and will update here when I do.

---
**Target Minecraft Versions:**     Any
**Requirements:**     None
**Related Issues:**     #2531

### CODE REVIEW

**Comment by Wealthyturtle** [comment]
Don't forget regression tests too.

**Comment by APickledWalrus** [comment]
Will make sure to add one in

**Comment by APickledWalrus** [comment]
That took more commits then it should've lol... forgot to fix organization :(

**Comment by bensku** [comment]
Not a regression, needs syntax text instead.

**Comment by APickledWalrus** [comment]
Okay, I’ll change this soon.

**Comment by APickledWalrus** [comment]
Changes made :)

File: src/main/java/ch/njol/skript/effects/EffReplace.java [hunk 1]
```diff
@@ -110,7 +126,12 @@ protected void execute(final Event e) {
 	
 	@Override
 	public String toString(final @Nullable Event e, final boolean debug) {
-		return "replace " + needles.toString(e, debug) + " in " + haystack.toString(e, debug) + " with " + replacement.toString(e, debug);
+		if (replaceFirst)
+			return "replace first " + needles.toString(e, debug) + " in " + haystack.toString(e, debug) + " with " + replacement.toString(e, debug)
+					+ "(case sensitive: " + caseSensitive + ")";
+		else
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/effects/EffReplace.java]: Minor nitpick, but you don't need an else here, since the other case is already handled.

Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/effects/EffReplace.java]: Fixed :)

**Pull request merged**